### DISCUSSION TITLE
Welcome to Octolapse Discussions!

### DISCUSSION THREAD

**Discussion by FormerLurker** [FormerLurker/Octolapse#647] [General]
## 👋 Welcome!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together 💪.

  To get started, comment below with an introduction of yourself and tell us about what you do with this community.

**Comment by joevj** [comment]
Hi there,

I'm rather new to 3D printing. I installed Octoprint on Ender 3 Pro. I installed OctoLapse. Now those small issues are fixed and working like a charm.

I have an led strip connected via relay. Relay is connected to GPIO 18. I can switch it on and off using the enclosure plugin.

What u want to do is to switch on the led before the snap is taken. I guess I can add some GCODE to the custom script of OctoLapse, before and after the photo is taken.

Can you help me with GCODE to be added too before and after to trigger the GPIO 18.

Thanks s lot

Regards
Joe

**Reply by FormerLurker** [reply]
This could be accomplished with a python script called by a bash script.  It's actually been done before :)  Working now, else I'd help more, but try searching through the issues for this.  I think there is some example script in there.

**Reply by joevj** [reply]
Thank you for feedback.

I was wondering if there is a GCODE to do it. I remember reading somewhere a GCODE to do it but can't find it now. Our may be I was mistaken.

I was expecting something like 
Gxx P18 1, to activate pin 18
Gxx P18 0, to deactivate the pin 18

May be in expecting too much 🙂

**Reply by MoX46** [reply]
There's a plugin called Gcode System Commands which allows you to define custom Gcode that executes local system commands. You can use this if you want to control via Gcode. 

The simpler solutions (imo) is to do what @FormerLurker said and use a bash script to execute a python script which controls the LED via GPIO. Check issue #334

**Reply by joevj** [reply]
Bit more googling landed in another discussion of yours..

I hope I can use below code. I need to figure out how to use this code as a script so that it can be called before snapshot is taken. A similar in to switch it off after.


```

#!/usr/bin/env python

import RPi.GPIO as GPIO
import time

GPIO.setwarnings(False)

GPIO.setmode(GPIO.BCM)
GPIO.setup(18, GPIO.OUT)

GPIO.output(21, True)

print "Done..."

```

**Reply by FormerLurker** [reply]
Thanks for the assist @joevj!

**Reply by joevj** [reply]
I tried this out and seems there is some issue..

i created two files using SSH. One for switching on the LED and another for Switching off.
File 1 : /home/pi/scripts/led-on.py

```

#!/usr/bin/env python

import RPi.GPIO as GPIO
import time

GPIO.setwarnings(False)

GPIO.setmode(GPIO.BCM)
GPIO.setup(18, GPIO.OUT)

GPIO.output(18, True)

print "Done..."
```


File 2 : /home/pi/scripts/led-off.py

```

#!/usr/bin/env python

import RPi.GPIO as GPIO
import time

GPIO.setwarnings(False)

GPIO.setmode(GPIO.BCM)
GPIO.setup(18, GPIO.OUT)

GPIO.output(18, False)

print "Done..."
```

**Updating Trigger scripts** 

I went back to the camera custom script anmd listed the files. but look like something is wrong..  snapshot is attached..  
![error](https://user-images.githubusercontent.com/46081408/103230748-35997380-495c-11eb-8bd9-10389725930c.png)

hope you help to know what i am doing wrong

**Reply by joevj** [reply]
sorry, I posted it too early..

I made below changes and is working..  awesome !!!

1.  added #!/bin/sh
2. removed the Print command.. not needed..
3. chmod +x to both files

it worked perfect then  :)

**Reply by FormerLurker** [reply]
@joevj, excellent!  Be sure to post a video of this working somewhere and link back here.  I love to see this stuff ;)

**Reply by joevj** [reply]
Definitely, will do that.

I just observed that the photo turned out to be blue. I may have to add a delay for led to be white colour. Will post the final config and a video.

**Reply by joevj** [reply]
> This could be accomplished with a python script called by a bash script. It's actually been done before :) Working now, else I'd help more, but try searching through the issues for this. I think there is some example script in there.

Hi @FormerLurker,
Thank you for the efforts for the Plugin.. its swesome..  thank you for your help as well..  I was able to switch on/off the LED using python script just before/after photo automatically. 

I consolidated the results in below blog post abd hope that helps others.  
could not get time to take video. .will do that soon..

(http://www.joevj.com/2021/04/controlling-led-through-octolapse-for.html)

**Reply by joevj** [reply]
Hi @FormerLurker 

Please check the video of the script for triggering relay working.. hope it is clear enough. Feel free to let me know if you have any feedback, would love to hear your feedback.

https://youtu.be/mYB_YnFkzhE


Thanks a lot
Joe

**Comment by joevj** [comment]
Hi FormerLurker/Octolapse

Please check the video
https://youtu.be/mYB_YnFkzhE


If you have feedback please let me know.

Thanks
Joe

**Reply by FormerLurker** [reply]
Hey, very cool!  I'll watch this more closely ASAP and will give you any feedback I come up with.  Thank you for sharing!

**Comment by joevj** [comment]
thank you


Best regards
Joe Varghese

Joe.Varghese.John@Gmail.com

**Comment by joevj** [comment]
Hi

If we have a provision to use Octolapse USB connection to just trigger the
camera, that would help. It's taking too long for image transfer. I could
render the photos manually into Timelapse then.

If we have a tweak it would be great🙏

Thanks
Joe

**Comment by silverdragonhawk** [comment]
Hi, I'm using Octoprint on windows 10 and I installed the mjpeg streamer from the microsoft store, and I get an image in the preview screen but I can't get it to work with Octolapse and when configuring the camera, I'm using a Logitech C922 Pro, the base, snapshot, and stream addresses don't work. I know with octopi you had to change a couple things in the txt file, but I don't know what or how to do that with octoprint installed on windows 10. I should also state that I'm in no way a coder so all of this is new to me. Thanks.

**Comment by 13rtn** [comment]
Hi, I'm fairly new to 3d printing but brand new with octoprint and octolapse. 

I need to learn, maybe I will ask you for information and explanations. 

Thanks in advance
